saleor/checkout/utils.py

```diff
@@ -429,13 +429,15 @@ def get_valid_shipping_methods_for_checkout(
     lines: Iterable[CheckoutLine],
     discounts: Iterable[DiscountInfo],
     country_code: Optional[str] = None,
+    prefetch_zip_codes: bool = False,
 ):
     manager = get_plugins_manager()
     return ShippingMethod.objects.applicable_shipping_methods_for_instance(
         checkout,
         channel_id=checkout.channel_id,
         price=manager.calculate_checkout_subtotal(checkout, lines, discounts).gross,
         country_code=country_code,
+        prefetch_zip_codes=prefetch_zip_codes,
```

Comment:
Not convinced about this pattern but don't have yet any idea how to fix it.

Reply:
applicable_shipping_methods_for_instance should prefetch it always, as we can make check of the zip codes inside it, would be even better if that method take it into consideration